src/components/app-navigator/TopTabNavigator.tsx

```diff
@@ -0,0 +1,218 @@
+    i += 1 / INDICATOR_ANIM_GRANULARITY
+  }
+
+  return {
+    inputRange,
+    outputRange
+  }
+}
+
+const createTabBarStyles = (themeColors: ThemeColors) =>
+  StyleSheet.create({
+    tabBarContainer: {
+      flexDirection: 'row',
+      paddingBottom: 4,
+      position: 'relative'
+    },
+
+    tabContainer: {
+      backgroundColor: themeColors.white,
+      flex: 1,
+      height: 48
+    },
+
+    tab: {
+      alignItems: 'center',
+      flex: 1,
+      flexDirection: 'column',
+      height: 48,
+      justifyContent: 'space-evenly'
+    }
+  })
+
+const TestTabBar = ({ state, descriptors, navigation, position }) => {
+  const screenWidth = Dimensions.get('screen').width
+  const isFocused = tabIndex => state.index === tabIndex
+  const styles = useThemedStyles(createTabBarStyles)
+
+  const onPress = (route, tabIndex: number) => {
+    const event = navigation.emit({
+      type: 'tabPress',
+      target: route.key,
+      canPreventDefault: true
+    })
+
+    if (!isFocused(tabIndex) && !event.defaultPrevented) {
+      // The `merge: true` option makes sure that the params inside the tab screen are preserved
+      navigation.navigate({ name: route.name, merge: true })
+    }
+  }
+
+  const onLongPress = (route, tabIndex: number) => {
+    navigation.emit({
+      type: 'tabLongPress',
+      target: route.key
+    })
+  }
+
+  const tabWidth = 100 / state.routes.length
+
+  const left = position.interpolate({
+    inputRange: [0, state.routes.length],
+    outputRange: [0, screenWidth]
+  })
+
+  const xScale = position.interpolate(
+    getSinAnimationRanges(state.routes.length)
+  )
+
+  return (
+    <View style={styles.tabBarContainer}>
+      {state.routes.map((route, index) => {
+        const { options } = descriptors[route.key]
+        const label = options.tabBarLabel ?? options.title ?? route.name
+        const Icon = options.tabBarIcon?.()
+        console.log({ options, label })
+
+        const inputRange = state.routes.map((_, i) => i)
+        const opacity = position.interpolate({
+          inputRange,
+          outputRange: inputRange.map(i => (i === index ? 1 : 0.52)) // opacity range
+        })
+
+        return (
+          <View key={label} style={styles.tabContainer}>
+            <TouchableOpacity
+              accessibilityLabel={options.tabBarAccessibilityLabel}
+              accessibilityRole='button'
+              accessibilityState={{ selected: isFocused(index) }}
+              onLongPress={() => onLongPress(route, index)}
+              onPress={() => onPress(route, index)}
+              style={styles.tab}
+              testID={options.tabBarTestID}
+            >
+              {Icon && (
+                <Animated.View style={{ opacity }}>
+                  <Icon />
+                </Animated.View>
+              )}
+              <Animated.Text
+                style={{
+                  color: '#858199',
```

Comment:
Does this correspond to a theme color? I think you could define these styles in the StyleSheet and then do something like:

``` 
<Animated.Text style={[styles.text, { opacity }]>
```